File: c++/stack.cpp
#include<iostream>
#include<stack>
using namespace std;

int main(){
    stack<int> mystack;
    mystack.push(10);
    mystack.push(20);
    cout<<".size() : "<<mystack.size()<<endl;
    cout<<".top() : "<<mystack.top()<<endl;
    mystack.pop();
    cout<<".top() : "<<mystack.top()<<endl;
    cout<<".empty() : "<<mystack.empty()<<endl;
    mystack.pop();
    cout<<".empty() : "<<mystack.empty()<<endl;
    // error => cout<<".top() : "<<mystack.top()<<endl;
}

File: PYTHON/stack.py
from collections import deque

mystack = deque()
mystack.append(10)
mystack.append(20)
print(".size() :", len(mystack))
print(".top() :", mystack[-1])  # Access the top element
mystack.pop()
print(".top() :", mystack[-1])  # Access the new top element
print(".empty() :", len(mystack) == 0)
mystack.pop()
print(".empty() :", len(mystack) == 0)
# Accessing the top of an empty stack will raise an IndexError
# Uncommenting the following line will cause an error:
# print(".top() :", mystack[-1])